eks-distro-base/Dockerfile.minimal-iptables

```diff
@@ -0,0 +1,62 @@
+# Copyright 2020 Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+ARG BASE_IMAGE
+ARG BUILDER_IMAGE
+FROM ${BUILDER_IMAGE} as builder
+
+RUN yum upgrade -y && \
+    yum update -y && \
+    yum install -y yum-utils
+
+ENV DOWNLOAD_DIR /tmp/download
+WORKDIR $DOWNLOAD_DIR
+
+RUN repotrack -p $DOWNLOAD_DIR \
+    ipset iptables kmod \
+    # Needed for conntrack-tools
+    libnetfilter_conntrack libnetfilter_cthelper libnetfilter_cttimeout libnetfilter_queue && \
+    # ebtables + conntrack-tools has a dep on systemd, which we do not need
+    # download them seperate and do not resolve dependencies
+    # deps are all downloaded above
+    yumdownloader --destdir=$DOWNLOAD_DIR ebtables conntrack-tools && \
+    # repotrack/yumdownloader will download multiple arch packages
+    rm -f *i686.rpm && \
+    # handled by minimal base
+    rm basesystem-* filesystem-* setup-* system-release-* tzdata* && \
+    # bash is a dep of glibc/common, but we do not want it in the image
+    # **** WE MAY NOT GET AWAY WITH THIS ****
+    rm bash-* ncurses-* && \
+    # chkconfig is used for update-alternatives whichw we do not need
+    rm chkconfig-*
+
+# not a perfect check, but sanity check to make sure we have downloaded
+# "all" deps except systemd
+# not perfect because a number of packages are already installed in the default image
+RUN yum install -y systemd && \
+    yum localinstall -y $DOWNLOAD_DIR/*
+
+WORKDIR /newroot
+
+RUN for rpm in $DOWNLOAD_DIR/*.rpm; do rpm2cpio $rpm | cpio -idmv; done && \
+    rm -rf usr/share/man usr/share/doc && \
+    # this is provided in minimal base, glibc brings its own, remove it
+    rm etc/nsswitch.conf
+
+# Store package list for future reference
+RUN find /tmp/download/*.rpm  -printf "%f\n" | sort >> /newroot/etc/packages && \
+    cat /newroot/etc/packages
```

Comment:
List of rpms which are extracted and placed on final image:

```#26 0.116 conntrack-tools-1.4.4-5.amzn2.2.x86_64.rpm
#26 0.116 ebtables-2.0.10-16.amzn2.0.1.x86_64.rpm
#26 0.116 glibc-2.26-48.amzn2.x86_64.rpm
#26 0.116 glibc-common-2.26-48.amzn2.x86_64.rpm
#26 0.116 glibc-minimal-langpack-2.26-48.amzn2.x86_64.rpm
#26 0.116 ipset-6.29-1.amzn2.0.1.x86_64.rpm
#26 0.116 ipset-libs-6.29-1.amzn2.0.1.x86_64.rpm
#26 0.116 iptables-1.8.4-10.amzn2.1.2.x86_64.rpm
#26 0.116 iptables-libs-1.8.4-10.amzn2.1.2.x86_64.rpm
#26 0.116 kmod-25-3.amzn2.0.2.x86_64.rpm
#26 0.116 libgcc-7.3.1-13.amzn2.x86_64.rpm
#26 0.116 libmnl-1.0.3-7.amzn2.0.2.x86_64.rpm
#26 0.116 libnetfilter_conntrack-1.0.6-1.amzn2.0.2.x86_64.rpm
#26 0.116 libnetfilter_cthelper-1.0.0-10.amzn2.1.x86_64.rpm
#26 0.116 libnetfilter_cttimeout-1.0.0-6.amzn2.1.x86_64.rpm
#26 0.116 libnetfilter_queue-1.0.2-2.amzn2.0.2.x86_64.rpm
#26 0.116 libnfnetlink-1.0.1-4.amzn2.0.2.x86_64.rpm
#26 0.116 libpcap-1.5.3-11.amzn2.x86_64.rpm
#26 0.116 libselinux-2.5-12.amzn2.0.2.x86_64.rpm
#26 0.116 libsepol-2.5-8.1.amzn2.0.2.x86_64.rpm
#26 0.116 libstdc++-7.3.1-13.amzn2.x86_64.rpm
#26 0.116 pcre-8.32-17.amzn2.0.2.x86_64.rpm
#26 0.116 popt-1.13-16.amzn2.0.2.x86_64.rpm
#26 0.116 xz-libs-5.2.2-1.amzn2.0.2.x86_64.rpm
#26 0.116 zlib-1.2.7-18.amzn2.x86_64.rpm
```

As you can see in the dockerfile, bash + systemd are technically required dependencies, but I think we can safely not include systemd and *maybe* we can get away with not having bash?  Apparently glibc-common requires bash because there is a script somewhere included in that package, my gut tells me this isnt actually going to try and be executed.